clients/web/react/src/_components/WebAuthnClient.ts

```diff
@@ -0,0 +1,566 @@
+import { get, create } from "@github/webauthn-json";
+import base64url from "base64url";
+import cbor from "cbor";
+import { Auth } from "aws-amplify";
+import axios from "axios";
+import credentialService from "../_services/credential.service";
+import { TrustedDeviceHelper } from "./TrustedDevices/TrustedDeviceHelper";
+
+import aws_exports from "../aws-exports";
+
+axios.defaults.baseURL = aws_exports.apiEndpoint;
+
+const WebAuthnClient = {
+  getAuthChallegeResponse,
+  getPublicKeyRequestOptions,
+  // getUsernamelessAuthChallegeResponse,
+  getUVFromAssertion,
+  sendChallengeAnswer,
+  signIn,
+  signUp,
+  registerNewCredential,
+};
+
+const defaultInvalidPIN = "-1";
+const webAuthnClientExceptionName = "WebAuthnClientException";
+const ERROR_CODE = "ERROR_CODE";
+const INVALID_CHALLENGE_TYPE = "INVALID_CHALLENGE_TYPE";
+
+function WebAuthnClientException(message, code = ERROR_CODE) {
+  const formattedError = {
+    message,
+    code,
+    name: webAuthnClientExceptionName,
+  };
+  return formattedError;
+}
+
+async function getPublicKeyRequestOptions(): Promise<any> {
```

Comment:
Suggest we comment each function to explain purpose, inputs, outputs, and exceptions in this module